example_keyspace_table_test.go

```diff
@@ -0,0 +1,60 @@
+package gocql_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/gocql/gocql"
+)
+
+// Example_keyspace_table demonstrate how Query.Keyspace() is updated based on prepared statement
+func Example_keyspace_table() {
+	/* The example assumes the following CQL was used to setup the keyspace:
+	create keyspace example with replication = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };
+	create table example.tweet(timeline text, id UUID, text text, PRIMARY KEY(id));
+	*/
+	cluster := gocql.NewCluster("localhost:9042")
+	cluster.Keyspace = "test"
+	cluster.Consistency = gocql.Quorum
+
+	fallback := gocql.RoundRobinHostPolicy()
+	cluster.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(fallback)
+
+	// connect to the cluster
+	session, err := cluster.CreateSession()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer session.Close()
+
+	ctx := context.Background()
+	id := gocql.TimeUUID()
+
+	// insert a tweet
+	if err := session.Query(`INSERT INTO example.tweet (timeline, id, text) VALUES (?, ?, ?)`,
+		"me", id, "hello world").WithContext(ctx).Exec(); err != nil {
+		log.Fatal(err)
+	}
+
+	var id2 gocql.UUID
+	var text string
+
+	/* Search for a specific set of records whose 'id' column matches
+	 * the value of inserted row. */
+	qry := session.Query(`SELECT id, text FROM example.tweet WHERE id = ? LIMIT 1`,
+		id).WithContext(ctx).Consistency(gocql.One)
+	if err := qry.Scan(&id2, &text); err != nil {
+		log.Fatal(err)
+	}
+
+	// cluster.Keyspace was set to "test", but during prepering statement
+	// Keyspace in Query should be changed to "example"
+	if qry.Keyspace() != "example" {
+		log.Fatal("The keyspace does not match the contents of the query.")
+	}
+
+	fmt.Println("Keyspace:", qry.Keyspace(), qry.Table())
+
+	// Keyspace: example tweet
```

Comment:
I don't get the purpose of this file. Is this an example to be ran by a user or a proper test? (if so, this should be a proper check rather than a `Println`)

Reply:
I have changed it to automated integration test.